django/core/files/uploadhandler.py

```diff
@@ -147,6 +155,15 @@ def file_complete(self, file_size):
         self.file.size = file_size
         return self.file
 
+    def upload_stopped(self):
+        temp_location = self.file.temporary_file_path()
+        try:
+            if os.name == 'nt':
+                self.file.close()
```

Comment:
Can you explain why we call it only on Windows?

Reply:
Trying to delete an open file on Windows raises a `PermissionError`, because it's [open by the same process that deletes it](https://github.com/django/django/pull/12475#issuecomment-673683214). Hence, we need to close the file before proceeding to delete it. On Posix systems, closing the file also deletes it. So to avoid, a `FileNotFoundError`, we close the file only on Windows.

Reply:
Thanks for explanation.